src/context.rs

```diff
@@ -913,7 +935,18 @@ impl Context {
         };
         let ret = Error::from_tss_rc(ret);
         if ret.is_success() {
-            Ok(ObjectHandle::from(esys_object_handle))
+            let new_object_handle = ObjectHandle::from(esys_object_handle);
+            // If you look at the specification and see that it says ESYS_TR_NULL
+            // then that is an error in the spec. ESYS_TR_NULL was renamed to
+            // ESYS_TR NONE.
+            if new_object_handle != ObjectHandle::None {
+                self.handle_manager
+                    .add_handle(new_object_handle, HandleDropAction::Close)?;
+            } else {
+                self.handle_manager.set_as_closed(object_handle)?;
+            }
```

Comment:
I'm guessing that before you can call `evict_control` you need to get an `ESYS_TR` handle, so you can't just call it without `tr_from_tpm_public`?

Reply:
Nah you could have a handle while running that you have evicted to make persistent and then you can evict the same handle again to make it transient again and when you do then the handles referring to that object becomes invalid. 

But you need to have an open and valid handle in order to call evict control when you want to make something transient.